doc/source/whatsnew/v0.24.0.txt

```diff
@@ -669,13 +669,16 @@ Indexing
 - Bug where indexing with a Numpy array containing negative values would mutate the indexer (:issue:`21867`)
 - Bug where mixed indexes wouldn't allow integers for ``.at`` (:issue:`19860`)
 - ``Float64Index.get_loc`` now raises ``KeyError`` when boolean key passed. (:issue:`19087`)
+- :class:`Index` no longer mangles ``None``, ``NaN`` and ``NaT``, i.e. they are treated as three different keys. However, for :class:`NumericIndex` all three are still coarsed to a ``NaN`` (:issue:`22332`)
+
 
 Missing
 ^^^^^^^
 
 - Bug in :func:`DataFrame.fillna` where a ``ValueError`` would raise when one column contained a ``datetime64[ns, tz]`` dtype (:issue:`15522`)
 - Bug in :func:`Series.hasnans` that could be incorrectly cached and return incorrect answers if null elements are introduced after an initial call (:issue:`19700`)
 - :func:`Series.isin` now treats all nans as equal also for `np.object`-dtype. This behavior is consistent with the behavior for float64 (:issue:`22119`)
+- :func:`unique` no longer manges NaN-float-values and ``NaT`` (:issue:`22295`)
```

Comment:
This new addition (unique not mangling different NAs) seems in constrast with `isin` actually treating them equal (the line above) ?

Reply:
The isin-fix is about `isin` treating different nan-float-object as the same entity, i.e.

 `algos.isin([float('nan')], [float('nan')])  # different nan-objects`

should yield `[True]` (and not [False]  as it is in 0.23).

The unique-fix is about treating `np.nan` and `pd.NaT` as two different entities, ie. 

`pd.unique([np.nan, pd.NaT])`

should yield `[np.nan, pd.NaT]` (and not `[np.nan]` as in 0.23). It still treats all nan-object as the same, i.e.:

`pd.unique([np.nan, float('nan'), -float('nan')] )   # three different nan-objects`

results in `[np.nan]`

I hope it is clearer with the updated more detailed description for the unique-change.